iso15118/shared/messages/iso15118_2/body.py

```diff
@@ -705,7 +712,7 @@ def get_message_name(self) -> str:
         """Returns the name of the one V2GMessage that is set for Body."""
         for k in self.__dict__.keys():
             if getattr(self, k):
-                return str(getattr(self, k))
+                return str(k)
```

Comment:
have you tested this? because this change will break the intention of this code...it will no longer print the class name but the key of the dictionary.
The dictionary of this class is like:
```python
{'current_demand_req': None, 'current_demand_res': CurrentDemandRes(....), 'metering_receipt_req*: None}`
````
And if you return `str(k)` will return 
`current_demand_res`

Reply:
Yes, it is tested and actually, this is what we want, you also can understand from the comment above and method name. Additionally, the return value will be used to find the message type in `get_msg_type` method which has the message names table.

Reply:
get_msg_type is defined as:
```python
def get_msg_type(msg_name: str) -> Optional[Type[BodyBase]]:
    """
    Returns the message type corresponding to the message name provided, or
    None if not match is found.

    Args:
        msg_name: The name of the message (e.g. SessionSetupReq)

    Returns: The message type corresponding to the given message name
    """
    msg_dict = {
        "SessionSetupReq": SessionSetupReq,
        "SessionSetupRes": SessionSetupRes,
        "ServiceDiscoveryReq": ServiceDiscoveryReq,
        "ServiceDiscoveryRes": ServiceDiscoveryRes,
        "ServiceDetailReq": ServiceDetailReq,
        "ServiceDetailRes": ServiceDetailRes,
        "PaymentServiceSelectionReq": PaymentServiceSelectionReq,
        "PaymentServiceSelectionRes": PaymentServiceSelectionRes,
        "CertificateInstallationReq": CertificateInstallationReq,
        "CertificateInstallationRes": CertificateInstallationRes,
        "PaymentDetailsReq": PaymentDetailsReq,
        "PaymentDetailsRes": PaymentDetailsRes,
        "AuthorizationReq": AuthorizationReq,
        "AuthorizationRes": AuthorizationRes,
        "CableCheckReq": CableCheckReq,
        "CableCheckRes": CableCheckRes,
        "PreChargeReq": PreChargeReq,
        "PreChargeRes": PreChargeRes,
        "ChargeParameterDiscoveryReq": ChargeParameterDiscoveryReq,
        "ChargeParameterDiscoveryRes": ChargeParameterDiscoveryRes,
        "PowerDeliveryReq": PowerDeliveryReq,
        "PowerDeliveryRes": PowerDeliveryRes,
        "ChargingStatusReq": ChargingStatusReq,
        "ChargingStatusRes": ChargingStatusRes,
        "CurrentDemandReq": CurrentDemandReq,
        "CurrentDemandRes": CurrentDemandRes,
        "MeteringReceiptReq": MeteringReceiptReq,
        "MeteringReceiptRes": MeteringReceiptRes,
        "WeldingDetectionReq": WeldingDetectionReq,
        "WeldingDetectionRes": WeldingDetectionRes,
        "SessionStopReq": SessionStopReq,
        "SessionStopRes": SessionStopRes,
    }
    return msg_dict.get(msg_name, None)
```

so, the key for checking it must be something like "SessionStopReq", but according with the mod you did, the key will be `session_stop_req` instead. Or am I missing something?

Reply:
any update on this @ikaratass

Reply:
You were right about this issue, the construct method of pydantic was creating different objects rather than expected, so i changed my implementation completely removing object creation without validation. we don't do that anymore. So this part of the code is reverted.